frontend/src/components/assessments/student-experience/question-elements/MultipleChoice.tsx

```diff
@@ -1,18 +1,30 @@
 import React from "react";
 import { Radio, RadioGroup, VStack } from "@chakra-ui/react";
 
+import { stringToNumberArray } from "../../../../utils/StudentUtils";
+
+import useAnswerState from "./useAnswerState";
+
 interface MultipleChoiceProps {
+  answerElementIndex: number;
   options: string[];
 }
+
 const MultipleChoice = ({
+  answerElementIndex,
   options,
 }: MultipleChoiceProps): React.ReactElement => {
+  const { currentAnswer, updateAnswer } = useAnswerState(answerElementIndex);
+
   return (
-    <RadioGroup>
+    <RadioGroup
+      onChange={(e) => updateAnswer(stringToNumberArray(e))}
+      value={currentAnswer.length > 0 ? currentAnswer[0].toString() : ""}
```

Comment:
```suggestion
      value={currentAnswer[0]?.toString() ?? ""}
```

is a bit more readable imo but up to you

Reply:
yes +1 much better, thank you!